src/dotnet-scaffolding/dotnet-scaffold-aspire/ScaffoldSteps/ValidateContextStep.cs

```diff
@@ -0,0 +1,34 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Scaffolding.Core.Scaffolders;
+using Microsoft.DotNet.Scaffolding.Core.Steps;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.Tools.Scaffold.Aspire.ScaffoldSteps;
+
+public class ValidateContextStep : ScaffoldStep
+{
+    public required Func<ScaffolderContext, ILogger, bool> ValidateMethod { get; set; }
+    private readonly ILogger _logger;
+
+    public ValidateContextStep(ILogger<ValidateContextStep> logger)
+    {
+        _logger = logger;
+    }
+
+    public override Task ExecuteAsync(ScaffolderContext context, CancellationToken cancellationToken = default)
+    {
+        if (!ValidateMethod(context, _logger))
+        {
+            _logger.LogError("Validation failed.");
+            return Task.FromResult(false);
```

Comment:
ScaffoldStep.ExecuteAsync doesn't actually return a value so that result doesn't go anywhere. We need to either change the signature to return Task<bool> (or some other way to indicate a result) or throw an exception. Probably Task<bool> I guess. And then we need to actually check the result in Scaffolder.ExecuteAsync.

Reply:
returning bool on steps now and checking for ContinueOnError as well